dashboard/app/models/user.rb

```diff
@@ -1252,6 +1252,26 @@ def can_edit_password?
     encrypted_password.present?
   end
 
+  def can_edit_display_name?
+    !picture_or_word_account?
+  end
+
+  def picture_or_word_account?
+    return false if sections_as_student.empty?
+    return false if encrypted_password.present?
+    sections_as_student.all? do |section|
+      section.login_type == Section::LOGIN_TYPE_PICTURE || section.login_type == Section::LOGIN_TYPE_WORD
+    end
```

Comment:
My opinion is that this isn't the right way to check for picture and word accounts. In particular, I think `provider == User::PROVIDER_SPONSORED` is both simpler and more correct. Your logic seems to fail in a variety of situations.
* If the account is sponsored and the sponsoring section was deleted.
* If the account is an oauth account belonging to picture sections (only).
* Perhaps others I'm not thinking of off the top of my head.